include/libtorrent/torrent_peer.hpp

```diff
@@ -213,12 +215,13 @@ namespace libtorrent {
 #if TORRENT_USE_I2P
 	struct TORRENT_EXTRA_EXPORT i2p_peer : torrent_peer
 	{
-		i2p_peer(char const* destination, bool connectable, peer_source_flags_t src);
-		i2p_peer(i2p_peer const&);
-		~i2p_peer();
-		i2p_peer& operator=(i2p_peer const&);
+		i2p_peer(string_view dst, bool connectable, peer_source_flags_t src);
+		i2p_peer(i2p_peer const&) = delete;
+		i2p_peer& operator=(i2p_peer const&) = delete;
+		i2p_peer(i2p_peer&&) = default;
+		i2p_peer& operator=(i2p_peer&&) = default;
 
-		char* destination;
+		string_ptr destination;
```

Comment:
question: what's the practical benefit of having this vs `std::string`? (since `string_ptr` copy the string)